### PR TITLE
Add random get mode in bobp

### LINKED ISSUE #215: Add random read mode to bobp

### PR DESCRIPTION
**Pull request by Justarone**
Closes #215

### CODE REVIEW

File: bob-tools/bin/bobp.rs [hunk 1]
```diff
@@ -76,6 +78,7 @@ struct TaskConfig {
     count: u64,
     payload_size: u64,
     direct: bool,
+    random: bool,
```
Comment (pyakushin) [inline bob-tools/bin/bobp.rs:81]: I suggest to use enum `Mode` for this purpose

File: bob-apps/bin/bobp.rs [hunk 2]
```diff
@@ -69,18 +71,35 @@ impl Debug for NetConfig {
...
     low_idx: u64,
     count: u64,
     payload_size: u64,
     direct: bool,
+    mode: Mode,
     measure_time: bool,
     key_size: usize,
 }
 
+fn unwrap_mode(mode: &str) -> Mode {
```
Comment (pyakushin) [inline bob-apps/bin/bobp.rs:91]: I suggest to impl Mode::from_str()

File: bob-apps/bin/bobp.rs [hunk 3]
```diff
@@ -69,18 +71,35 @@ impl Debug for NetConfig {
...
+    match mode.as_ref() {
+        "normal" => Mode::Normal,
+        "random" => Mode::Random,
+        _ => Mode::Normal,
```
Comment (pyakushin) [inline bob-apps/bin/bobp.rs:95]:
```rust
if mode == "random" {
    Mode::Random
} else {
    Mode::Normal
}
```

Comment (pyakushin) [reply bob-apps/bin/bobp.rs:95]: Or add panic if Mode doesn't match `normal` or `random`

**Review by pyakushin — APPROVED** [review]

File: bob-apps/bin/bobp.rs [hunk 4]
```diff
@@ -737,6 +772,12 @@ fn get_matches() -> ArgMatches<'static> {
                 .takes_value(false)
                 .long("verify"),
         )
+        .arg(
+            Arg::with_name("mode")
+                .help("keys in get operation are shuffled")
+                .takes_value(false)
```
Comment (ikopylov) [inline bob-apps/bin/bobp.rs:778]: Now `mode` takes one of two values: `normal` or `random`, so the `true` should be passed. Additionally, default value should be specified

**Review by ikopylov — APPROVED** [review]
LGTM

**Pull request merged**